### PR TITLE
Updates the save and analytics of section setup

### PR DESCRIPTION
**Pull request by hannahbergam**
This pr might look like a lot of changes but I should be able to summarize them in a brief way:

1. It re-enables the save testing file!
2. It moves the analytics reporting to only call 'section setup completed' if the save actually goes through (instead of sending a complete message even when the save gives an error
3. It updates the error messaging and the form errors to set the save state back to false so a user can read the error, correct, and try again
4. It moves the save function back inside the component for cleanliness

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/TEACH-465

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 1]
```diff
@@ -186,6 +134,65 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
...
+    })
+      .then(response => response.json())
+      .then(data => {
+        recordSectionSetupEvent(section);
+        // Redirect to the sections list.
+        window.location.href = window.location.origin + '/home';
+      })
+      .catch(err => {
+        setIsSaveInProgress(false);
+        console.error(err);
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:206]: I checked how we do error messaging in other files, and this is a standard pattern. Happy to hear input on better/more helpful messaging!

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 2]
```diff
@@ -186,6 +134,65 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
...
+      : section.loginType;
+    const participantType = isNewSection
+      ? queryParams('participantType')
+      : section.participantType;
+
+    const form = document.querySelector(`#${FORM_ID}`);
+    // If we find a missing field in the form, report which one and reset save status
+    if (!form.checkValidity()) {
+      form.reportValidity();
+      setIsSaveInProgress(false);
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:162]: This is the validation that catches when a user doesn't fill out a required field. The only change in this area: added the setIsSaveInProgress(false).

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 3]
```diff
@@ -186,6 +134,65 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
     }
   };
 
+  const saveSection = section => {
+    // Determine data sources and save method based on new vs edit section
+    const dataUrl = isNewSection
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:147]: This first part is all the same- just a clean copy/paste

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx [hunk 4]
```diff
@@ -291,8 +298,7 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
           onClick={e => {
             e.preventDefault();
             setIsSaveInProgress(true);
-            recordSectionSetupEvent(sections[0]);
```
Comment (hannahbergam) [inline apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:312]: I moved this call into the save function right before the redirect to ensure the event is a true successful save

**Comment by tess323** [comment]
lurking 👀 I am fine with making this work with oauth being follow up

File: apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx:206]: This seems fine for now, though it wouldn't give any feedback to the user. We should probably (in a followup) figure out a way to show them the error if saving fails

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**